spec/app/controllers/courses_controller/controller_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+RSpec.describe CoursesController, type: :request do
+  let(:request_payload) { 
+    {
+      course_uuid: given_course_data.uuid,
+      ecosystem_uuid: given_course_data.ecosystem_uuid
+    }
+  }
+
+  let(:given_course_data) {
+    build(:course, ecosystem_uuid: ecosystem.uuid)
+  }
+
+  let(:ecosystem) {
+    build(:ecosystem)
+  }
+
+  let(:target_result) {
+    {
+      created_course_uuid: given_course_data.uuid
+    }
```

Comment:
This is purely a style thing, but one line is fine for one-pair hashes: `{ created_course_uuid: given_course_data.uuid }`